src/main/java/uk/gov/ons/census/action/schedule/ActionInstructionBuilder.java

```diff
@@ -0,0 +1,119 @@
+package uk.gov.ons.census.action.schedule;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.action.model.dto.instruction.ActionAddress;
+import uk.gov.ons.census.action.model.dto.instruction.ActionEvent;
+import uk.gov.ons.census.action.model.dto.instruction.ActionInstruction;
+import uk.gov.ons.census.action.model.dto.instruction.ActionRequest;
+import uk.gov.ons.census.action.model.dto.instruction.Priority;
+import uk.gov.ons.census.action.model.entity.ActionRule;
+import uk.gov.ons.census.action.model.entity.Case;
+import uk.gov.ons.census.action.model.entity.UacQidLink;
+import uk.gov.ons.census.action.model.repository.UacQidLinkRepository;
+
+@Component
+public class ActionInstructionBuilder {
+  private final UacQidLinkRepository uacQidLinkRepository;
+
+  @Value("${queueconfig.outbound-exchange}")
+  private String outboundExchange;
+
+  public ActionInstructionBuilder(UacQidLinkRepository uacQidLinkRepository) {
+    this.uacQidLinkRepository = uacQidLinkRepository;
+  }
+
+  public ActionInstruction buildActionInstruction(Case caze, ActionRule actionRule) {
+
+    List<UacQidLink> uacQidLinks = uacQidLinkRepository.findByCaseId(caze.getCaseId().toString());
+
+    UacQidLink uacQidLink;
+    Optional<UacQidLink> uacQidLinkWales = Optional.ofNullable(null);
+
+    if (uacQidLinks == null || uacQidLinks.isEmpty()) {
```

Comment:
I know this code is copy and pasted from elsewhere but is it possible to move this functionality off into clearly named functions.  Something like getUacQidLink and getWelshUacQidLink, with the same sort of gubbins.  Would make it this function less noisy.